api/envoy/config/filter/udp/dns_filter/v2alpha/dns_filter.proto

```diff
@@ -0,0 +1,64 @@
+syntax = "proto3";
+
+package envoy.config.filter.udp.dns_filter.v2alpha;
+
+import "google/protobuf/duration.proto";
+
+import "udpa/annotations/status.proto";
+
+import "udpa/annotations/migrate.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.config.filter.udp.dns_filter.v2alpha";
+option java_outer_classname = "DnsFilterProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_migrate).move_to_package =
+    "envoy.extensions.filters.udp.dns_filter.v3";
+option (udpa.annotations.file_status).work_in_progress = true;
+
+// Configuration for the DNS filter.
+message DnsFilterConfig {
+  // Specify the method of determining the address of a given endpoint
+  enum EndpointType { STATIC = 0; }
+
+  // This message type is extensible and can contain a list of addresses
+  // or dictate some other method for resolving the addresses for an
+  // endpoint
+  message DnsEndpoint {
+    EndpointType type = 1;
+
+    // A list of IP addresses Envoy returns to queries for the given domain
+    repeated string address = 2;
+  }
+
+  message DnsVirtualDomain {
+    // The domain name for which Envoy will respond to query requests
+    string name = 1 [(validate.rules).string = {min_len: 2}];
```

Comment:
@asraa what PGV constraints should be here?

Reply:
Currently, it looks like `name` is used to populate the string field in this map https://github.com/envoyproxy/envoy/blob/92bee86cf3616571054e908ebbbffac58148af87/source/extensions/filters/udp/dns_filter/dns_filter.h#L36

So no conditions are enforced. 

@abaptiste do you anticipate needing to use `LowerCaseString` like how domain names are treated here? https://github.com/envoyproxy/envoy/blob/55971b23978b961da71c80b49ddede4c88f24eab/source/common/router/config_impl.cc#L1072 If so, we now have non-strict `HTTP_HEADER_VALUE` checks that match the checks in `LowerCaseString`. 

Usage looks like:
`string x = 1 [(validate.rules).string {well_known_regex: HTTP_HEADER_VALUE, strict: false}];`

The PR that updates the PGV dep for that option is out now though.

Reply:
Good point.  I have been assuming the names are lower case but enforcing that is a good idea.   I will add this to the validation.